<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CreateProofDialog</class>
 <widget class="QDialog" name="CreateProofDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>286</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>450</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Create proof</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="label">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>User message:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="messageEdit">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="placeholderText">
      <string>Entered message will be embedded to proof</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="addressLabel">
     <property name="text">
      <string>Address:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QComboBox" name="addressesBox">
     <property name="duplicatesEnabled">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="proofLabel">
     <property name="text">
      <string>Proof:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTextEdit" name="proofEdit">
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="placeholderText">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="generateButton">
       <property name="text">
        <string>Generate</string>
       </property>
       <property name="autoDefault">
        <bool>true</bool>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="copyButton">
       <property name="text">
        <string>Copy to clipboard</string>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>0</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Close</string>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>CreateProofDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>236</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>copyButton</sender>
   <signal>clicked()</signal>
   <receiver>CreateProofDialog</receiver>
   <slot>copyToClipboard()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>236</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generateButton</sender>
   <signal>clicked()</signal>
   <receiver>CreateProofDialog</receiver>
   <slot>generateProof()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>343</x>
     <y>236</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addressesBox</sender>
   <signal>activated(int)</signal>
   <receiver>CreateProofDialog</receiver>
   <slot>addressChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>196</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addressesBox</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>CreateProofDialog</receiver>
   <slot>currentAddressChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>196</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>generateProof()</slot>
  <slot>copyToClipboard()</slot>
  <slot>addressChanged(int)</slot>
  <slot>currentAddressChanged()</slot>
 </slots>
</ui>
